src/imageio/storage/disk.c

```diff
@@ -111,6 +118,42 @@ void *legacy_params(dt_imageio_module_storage_t *self, const void *const old_par
     *new_size = self->params_size(self);
     return n;
   }
+
```

Comment:
I don't think you need a `legacy_params()` routine at all as you don't change `dt_imageio_module_storage_t` definition.

Reply:
@Sturmflut : Any news on this?

Reply:
Hi, yes, sorry, I'm short on time. I added code to `legacy_params` because I changed the order of the options in the `dt_disk_onconflict_actions_t`  enum. `DT_EXPORT_ONCONFLICT_SKIP` now has a different index than before. The code in `legacy_params` is not strictly necessary because the number of indexes has increased, but without the conversion routine if someone set the setting to "skip", it suddenly changes to "overwrite if changed" after the update, which completely changes the behaviour on export and might be very unsuspected and hard to diagnose for the user.